dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -282,6 +282,13 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
         List<Workflow> workflows = workflowDAO.findAllByPath("github.com/" + repository, false).stream().filter(workflow -> Objects.equals(workflow.getMode(), DOCKSTORE_YML)).collect(
                 Collectors.toList());
 
+        // When the git reference is the default version, unset the default version
+        workflows.forEach(workflow -> {
```

Comment:
This may depend on the resolution to Louise's comment, but you should probably make this a `filter/findFirst` so you don't keep looping through the versions even after you've found the default.

Reply:
This is looping over workflows, not versions, since a git repo can have more than 1 workflow.
Different workflows can have different default versions.

Reply:
D'oh!